src/basic/hash-funcs.c

```diff
@@ -36,6 +37,55 @@ const struct hash_ops string_hash_ops = {
         .compare = string_compare_func
 };
 
+
+void path_hash_func(const void *p, struct siphash *state) {
```

Comment:
Why do not you use normalize_path()? This is just a knee-leg reaction. Please ignore if it does not make sense.

Reply:
yes, we could do that, and so far we usually did. But I think it's prefereable if we don't require such preparations and things would just work without having to think about this.

There are two big benefits over doing path-aware hashing like this patch adds it: normalize_path() won't remove trailing slashes, but for the hash tables here we will consider "foo/" and "foo" a key collision (which I think it's a good thing). More importantly though: note that we'd have to call normalize_path() not only when adding something to a hashtable but also when looking something up in it. And that gets quickly annoying, as whenever we deal with const strings we'd first have to copy the string, then normalize it and then call hashmap_get(). With this new logic all this goes away, you can just use any path you like and it will just work.

Moreover, I think there's value if the user of the hashmap can put stuff in it and be somewhat sure he gets out of it again what he put in...

Reply:
Thank you for the detailed explanation.